### PR TITLE
[updates] fixes from style lib removal

### PR DESCRIPTION
**Pull request by phillimorland**
Mostly fixes https://github.com/liqd/a4-meinberlin/issues/4515
fixes https://github.com/liqd/a4-meinberlin/issues/4521

### CODE REVIEW

**Review by khamui — COMMENTED** [review]
really nice! few comments :)

**Review by fuzzylogic2000 — APPROVED** [review]
Looks good! One tiny question, though.

File: meinberlin/assets/scss/components/_button.scss [hunk 1]
```diff
@@ -39,29 +7,28 @@
...
 
-.btn--flex {
-    @extend %button-base;
-    @include button-color($link-color-btn, $link-hover-color-btn);
-    flex: 1 0 auto;
-    height: 100%;
-
-    &.bottom {
-        align-self: flex-end;
+    &.is-disabled {
```
[the remark this answers is not in the document]
Comment (fuzzylogic2000) [reply meinberlin/assets/scss/components/_button.scss:10]: Ok, I answered that myself, yes we do.

[the remark this answers is not in the document]
Comment (phillimorland) [reply meinberlin/assets/scss/components/_button.scss:10]: it is still used in some old code unfortunately also for links

File: meinberlin/assets/scss/components/_timeline.scss [hunk 2]
```diff
@@ -62,10 +62,10 @@ $circle-size: 0.8em;
     }
 
     &.is-selected {
-        @include button-color($link-color, $link-hover-color);
+        // @include button-color($link-color, $link-color, $link-color, $link-hover-color);
```
[the remark this answers is not in the document]
Comment (phillimorland) [reply meinberlin/assets/scss/components/_timeline.scss:65]: no I just forgot

File: meinberlin/assets/scss/components/_button.scss [hunk 3]
```diff
@@ -28,15 +29,13 @@
...
-    // @include button-color($border-color, transparent, transparent, transparent);
-
     &,
     &:focus,
     &:hover,
     &:active,
     &:disabled,
     &.is-disabled {
-        border-color: $border-color;
+        border-color: $border-color !important;
```
[the remark this answers is not in the document]
Comment (phillimorland) [reply meinberlin/assets/scss/components/_button.scss:38]: it's being overwritten in BS which wasn't happening before, I also think we should maybe phase out this button anyway as it's not very accessible, will check how extensively it is used

[the remark this answers is not in the document]
Comment (phillimorland) [reply meinberlin/assets/scss/components/_button.scss:38]: ok used quite alot and I need a little time away from css but will come back to it

File: meinberlin/assets/scss/components/_button.scss [hunk 4]
```diff
@@ -1,6 +1,5 @@
-
 %button-base {
-    @extend %input-base;
+    @extend %input-base; // needed for stretched dropdowns ect.
```
[the remark this answers is not in the document]
Comment (phillimorland) [reply meinberlin/assets/scss/components/_button.scss:2]: @khamui it's a good point, I had a think and can't think of a better name then interaction-base and that doesn't sit right somehow, will keep thinking but for the time being I have moved it into base.scss so by context it is clearer that it is for various types of input and will try update name to something that feels better

[the remark this answers is not in the document]
Comment (khamui) [reply meinberlin/assets/scss/components/_button.scss:2]: cool! hmm..i tried to find another word and looked up mozilla how they name these kind of elements. Button is an interactive element, but input and textareas etc. are not.

Maybe `control-base`? If you feel there is no better word we could at least add a comment describing it, just to help us understanding next time?

**Pull request merged**